### ISSUE TITLE
Wrong indentation: Parser error lacking description

### ISSUE THREAD

**Issue by Abastro** [haskell/error-messages#31]
Code in question:
```hs
let i = 1
  j = 2
  in ...
```

Error message:
```
site.hs:104:3: error: parse error on input ‘j’
    |
104 |   j = 2
    |   ^
```
(there are 103 lines above this let clause, but surely those parts are irrelevant)

Suggested Improvements:
```
site.hs:104:3: error: parser error on input 'j'
`let` clause require indenting at least 3 spaces
103 | let i = 1
104 |   j = 2
    |   ^
```

Or, at least
`error: parser error on input 'j' (likely incorrect indentation)`

There should be similar class of errors, but let-clause related ones were the most frequent and problematic.

**Comment by Abastro** [comment]
I like your wording! That should be better.
How much effort would it require to implement better parsing?
Anything I can help? (Tho I doubt I could help with core parts of ghc)

**Comment by Ericson2314** [comment]
Yeah the way we do layout today is....disgusting, but I have very little clue how to do better, and it would be a big breaking change and costly migration period.

I would love to if someone could do (masters? PHD?) thesis, though.

**Comment by Ericson2314** [comment]
Awesome! Glad to hear it. If there is anything that needs to be done to happy, well, maybe I can help make it happen :)
